lib/OpenQA/Git/ServerAvailability.pm

```diff
@@ -0,0 +1,42 @@
+# Copyright SUSE LLC
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+package OpenQA::Git::ServerAvailability;
+use Mojo::Base -strict, -signatures;
+
+use Mojo::File qw(path);
+use Time::Seconds qw(ONE_HOUR);
+use OpenQA::Utils qw(prjdir);
+use Exporter qw(import);
+
+our @EXPORT_OK = qw(report_server_unavailable report_server_available SKIP FAIL);
+
+use constant MAX_DURATION_SECONDS => ($ENV{OPENQA_GIT_SERVER_OUTAGE_DURATION} || (ONE_HOUR / 2));
+
+use constant {
+    SKIP => 'SKIP',
+    FAIL => 'FAIL',
+};
+
+sub _state_file ($app, $server_name) {
+    return path(
+        ($ENV{OPENQA_GIT_SERVER_OUTAGE_FILE} // ($app->config->{global}->{base_dir} // prjdir) . '/git_server_outage')
```

Comment:
And use `path($foo, $bar)` to concatenate a valid path. See https://docs.mojolicious.org/Mojo/File#path